__tests__/index.js

```diff
@@ -455,4 +460,31 @@ describe('create.ImportDeclaration', () => {
     expect(getFilename).toBeCalledTimes(1)
     expect(report).not.toBeCalled()
   })
+
+  it('should not report if allowTypeImport is true', () => {
+    resolveImportPath.mockReturnValue('src/components/ui/Text')
+    const getFilename = jest.fn(() =>
+      path.join(process.cwd(), 'src/components/ui/aaa.ts')
```

Comment:
I noticed that this parameter seems to depend on the behavior of allowSameModule. 
Just to be on the safe side, could we change it to import from a path outside of the module?

```suggestion
      path.join(process.cwd(), 'src/pages/index.tsx')
```